src/App.tsx

```diff
@@ -15,47 +16,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleListReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const handleListSortByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const handleListSortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const handleListSortReset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+
+  visibleGoods.sort((goodA, goodB) => {
```

Comment:
> function is not complete

move this flow to the function

Reply:
thx! console.log confused me and I misunderstood how it has to work